### DISCUSSION TITLE
How to pass GET parameters of gateway endpoint to POST backend method as body `application/x-www-form-urlencoded`?

### DISCUSSION THREAD

**Discussion by ernoli** [ThreeMammals/Ocelot#1946] [Q&A] [answered]
Hello

I am sorry I am writing right here, I just can't find much of community resources for Ocelot save from some stackoverflow 

So here's question

My target API backend methods expects parameters to be passed in body as content-type of ```application/x-www-form-urlencoded```

So, basically I need to transform my GET request to my gateway endpoint ```/gw/test/?param1={param1}&param={param2}``` to POST backend request to endpoint ```/api/test``` with parameters sent as body 

```param1={param1}&param2={param2}```

and I just can't figure out is it even possible with Ocelot right now?

**Comment by garybond** [comment, marked as answer]
I've had to do the same thing recently to take a GET request with parameters and assemble some json to be POSTed to the back-end service, 
I ended up using the [DelegatingHandler](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html).

This should get you off in the right direction...
```csharp
public class ConvertGetToPostHandler : DelegatingHandler
{
    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
    {
        //do stuff to get/validate your querystring parameters, e.g.
        var yourFormData = request.RequestUri.Query.TrimStart('?');

        request.Content = new StringContent(yourFormData, Encoding.UTF8, "application/x-www-form-urlencoded");
        request.Method = HttpMethod.Post;

        return await base.SendAsync(request, cancellationToken);
    }
}
```

Of course, I'd be happy to get another opinion...

**Reply by raman-m** [reply]
@ernoli  What about this solution?
